src/routes/auth/user/verify.ts

```diff
@@ -0,0 +1,68 @@
+import { REDIRECT_URL_ERROR, REDIRECT_URL_SUCCESS } from '@shared/config'
+import { Request, Response } from 'express'
+import { activateUser, changeEmailByTicket } from '@shared/queries'
+
+import Boom from '@hapi/boom'
+import { asyncWrapper } from '@shared/helpers'
+import { request } from '@shared/request'
+import { v4 as uuidv4 } from 'uuid'
+import { verifySchema } from '@shared/schema'
+
+interface HasuraData {
+  update_users: { affected_rows: number }
+}
+
+async function verifyUser({ query }: Request, res: Response): Promise<unknown> {
+  let hasuraData: HasuraData
+
+  /**
+   * Delete undefined query components.
+   */
+  Object.keys(query).forEach(k => query[k] === 'undefined' && delete query[k])
+
+  const { ticket, new_email } = await verifySchema.validateAsync({
+    ticket: query.ticket,
+    new_email: query?.new_email
```

Comment:
I think this is a security flaw.

A user can send *any* email in `new_email` to update their email to whatever they want?

Proposal:
Save the `new_email` in the database. Create a separata endpoint to update from `email` to `new_email` based on the ticket.

Reply:
Good idea, I'll fix that!